subsys/net/lib/aws_iot/src/aws_iot.c

```diff
@@ -535,7 +535,7 @@ static int publish_get_payload(struct mqtt_client *const c, size_t length)
 static void mqtt_evt_handler(struct mqtt_client *const c,
 			     const struct mqtt_evt *mqtt_evt)
 {
-	int err;
+	int err = 0;
```

Comment:
I see that there is one place where this variable could be used w/o initialization (at line 586), but it seems it's used wrongly in there (`res` should be used instead, or we could even give up on `res` variable and do `err = topic_subscribe();`, if I'm not mistaken). IMO it should be fixed in there instead.